### PR TITLE
Issue/54 doughnut

### LINKED ISSUE #54: Re-implement doughnut dispatch verification
Add cennznut verification back into CENNZnet. It was removed from the `develop` branch to simplify the upgrade process.

Tasks:
- [ ] Re-implement Doughnut/CENNZnut dispatch verification by adding back in: https://github.com/cennznet/cennznet/blob/master/runtime/src/doughnut.rs and update dependencies

- [x] Replace the plug impl:
https://github.com/cennznet/cennznet/blob/47b94c97191ec2ef4d3f50de9f31d7a9b3cd70e4/runtime/src/lib.rs#L126

### PR DESCRIPTION
**Pull request by mzxyz**
Closes #54

### CODE REVIEW

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -473,6 +480,29 @@ impl system::offchain::CreateTransaction<Runtime, UncheckedExtrinsic> for Runtim
...
+	const DOMAIN: &'static str = "cennznet";
+
+	fn verify_dispatch(doughnut: &CennznetDoughnut, module: &str, method: &str) -> Result<(), &'static str> {
+		let mut domain = doughnut
+			.get_domain(Self::DOMAIN)
+			.ok_or("Doughnut does not grant permission for cennznet domain")?;
+		let cennznut: CENNZnutV0 = Decode::decode(&mut domain).map_err(|_| "Bad Doughnut encoding")?;
+
+		// Strips [c|p|s]rml- prefix
+		let module = cennznut
```
Comment (jordy25519) [inline runtime/src/lib.rs]: Should use cennzut.validate() here (via trait Validate). Can just pass empty `args` for now `&[]` https://github.com/cennznet/cennznut-rs/blob/e09a993add8a831ecb574e068e730df76ac8c757/src/lib.rs#L29

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -25,6 +25,7 @@ use babe_primitives::{AuthorityId as BabeId, AuthoritySignature as BabeSignature
 use cennznet_primitives::{
 	AccountId, AccountIndex, AssetId, Balance, BlockNumber, Doughnut, Hash, Index, Moment, Signature,
 };
+use cennznut::CENNZnutV0;
```
Comment (jordy25519) [inline runtime/src/lib.rs]: Is this import needed?

**Review by jordy25519 — CHANGES_REQUESTED** [review]
Nice work. Should change to use the cennznut validate trait and I think it's good to go

File: runtime/src/lib.rs (hunk 2 above)
Comment (mzxyz) [reply runtime/src/lib.rs]: Yes, https://github.com/cennznet/cennznet/blob/85f69d8041e9deb0b8d9804a868800c1460f2d3f/runtime/src/lib.rs#L491

**Review by cowboy-bebug — APPROVED** [review]
Really nice work! 😄

**Review by jordy25519 — COMMENTED** [review]
Just checking the lock file. I think it's out of date.
pest shouldn't need to be pulled in any longer 
https://github.com/cennznet/cennznet/pull/81/files#diff-d2ede298d9a75c849170d6ef285eda1eR2620-R2621

---
Update: seems to be fine

**Review by jordy25519 — APPROVED** [review]

**Pull request merged**